apis/discoveryengine/v1alpha1/datastore_types.go

```diff
@@ -15,65 +15,178 @@
+	// +kubebuilder:validation:Enum=THIRD_PARTY;GSUITE;IDP_TYPE_UNSPECIFIED
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.IdpConfig.idp_type
+	IdpType *string `json:"idpType,omitempty"`
 
-	// Immutable. The collection for the DataStore.
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Collection field is immutable"
-	// +required
-	Collection string `json:"collection"`
+	// External Identity provider config.
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.IdpConfig.external_idp_config
+	ExternalIdpConfig *IdpConfig_ExternalIdpConfig `json:"externalIdpConfig,omitempty"`
+}
 
-	// Immutable. The location for the resource.
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location field is immutable"
-	// +required
-	Location string `json:"location"`
+// +kcc:proto=google.cloud.discoveryengine.v1alpha.DocumentProcessingConfig
+type DocumentProcessingConfig struct {
+	// Whether chunking mode is enabled.
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.DocumentProcessingConfig.chunking_config
+	ChunkingConfig *DocumentProcessingConfig_ChunkingConfig `json:"chunkingConfig,omitempty"`
+
+	// Configurations for default Document parser.
+	//  If not specified, we will configure it as default DigitalParsingConfig, and
+	//  the default parsing config will be applied to all file types for Document
+	//  parsing.
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.DocumentProcessingConfig.default_parsing_config
+	DefaultParsingConfig *DocumentProcessingConfig_ParsingConfig `json:"defaultParsingConfig,omitempty"`
+
+	// TODO: unsupported map type with key string and value message
+}
+
+// +kcc:proto=google.cloud.discoveryengine.v1alpha.LanguageInfo
+type LanguageInfo struct {
+	// The language code for the DataStore. See https://cloud.google.com/vertex-ai/docs/general/supported-languages
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.LanguageInfo.language_code
+	LanguageCode *string `json:"languageCode,omitempty"`
+}
+
+// +kcc:proto=google.cloud.discoveryengine.v1alpha.DataStore
+type DataStoreObservedState struct {
+	// Output only. The id of the default
+	//  [Schema][google.cloud.discoveryengine.v1alpha.Schema] asscociated to this
+	//  data store.
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.DataStore.default_schema_id
+	DefaultSchemaID *string `json:"defaultSchemaID,omitempty"`
+
+	// Output only. Timestamp the
+	//  [DataStore][google.cloud.discoveryengine.v1alpha.DataStore] was created at.
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.DataStore.create_time
+	CreateTime *string `json:"createTime,omitempty"`
+
+	// Language info for DataStore.
+	// +kcc:proto:field=google.cloud.discoveryengine.v1alpha.DataStore.language_info
+	LanguageInfo *LanguageInfoObservedState `json:"languageInfo,omitempty"`
+}
+
+// +kcc:proto=google.cloud.discoveryengine.v1alpha.WorkspaceConfig
+type WorkspaceConfig struct {
```

Comment:
This one looks like it should have been auto-generated also

Reply:
The proto comments tell nothing about this value. I pulled it out from the proto source code.  

Oh, do you mean having the type-generator to generate this enum validator? That could be a good idea!